drivers/npmx/npmx_shell.c

```diff
@@ -2367,6 +2367,47 @@ static int cmd_leds_state_set(const struct shell *shell, size_t argc, char **arg
 	return 0;
 }
 
+static int cmd_gpio_status_get(const struct shell *shell, size_t argc, char **argv)
+{
+	npmx_instance_t *npmx_instance = npmx_driver_instance_get(pmic_dev);
+
+	if (npmx_instance == NULL) {
+		shell_error(shell, "Error: shell is not initialized.");
+		return 0;
+	}
+
+	if (argc < 2) {
+		shell_error(shell, "Error: missing GPIO number.");
+		return 0;
+	}
+
+	int err = 0;
+	uint8_t gpio_idx = CLAMP(shell_strtoul(argv[1], 0, &err), 0, UINT8_MAX);
+
+	if (err != 0) {
+		shell_error(shell, "Error: GPIO number has to be an integer.");
+		return 0;
+	}
+
+	if (gpio_idx >= NPM_GPIOS_COUNT) {
+		shell_error(shell, "Error: GPIO number is too high: no such instance.");
+		return 0;
+	}
+
+	npmx_gpio_t *gpio_instance = npmx_gpio_get(npmx_instance, gpio_idx);
+
+	bool gpio_status;
+	npmx_error_t err_code = npmx_gpio_status_check(gpio_instance, &gpio_status);
+
+	if (!check_error_code(shell, err_code)) {
+		shell_error(shell, "Error: unable to read GPIO status.");
+	} else {
+		shell_print(shell, "Value: %u.", (uint8_t)gpio_status);
```

Comment:
```suggestion
		shell_print(shell, "Value: %d.", gpio_status);
```

Reply:
As there are no exact guidelines in the coding standard regarding the casting of bool variables, I leave it as it is to be compatible with previous comments:
https://github.com/NordicSemiconductor/npmx-zephyr/pull/89#discussion_r1347390067